src/react/contexts/room-context.tsx

```diff
@@ -0,0 +1,8 @@
+import { Room } from '@ably/chat';
+import { createContext } from 'react';
+
+interface RoomContextType {
```

Comment:
Spoke briefly in standup today but I thought I'll answer here to have a record too.

I saw little added value from putting them together. If we use `<RoomProvider>` anywhere, it must be inside a `<ChatClientProvider>` otherwise `useChatClient` won't work. The rooms themselves are already singletons due to how `client.rooms` works, there's close to no extra benefits if we mix the contexts together. (that, or I'm missing something, still learning react and it still surprises me)